socorro/external/es/crashstorage.py

```diff
@@ -168,13 +184,39 @@ def reconstitute_datetimes(processed_crash):
                 # not there? we don't care
                 pass
 
+    @staticmethod
+    def remove_bad_keys(raw_crash):
+        """Removes keys from the top-level of the dict that are bad
+
+        Good keys satisfy the following properties:
+
+        * have one or more characters
+        * are composed of a-zA-Z0-9_-
+
+        Anything else is a bad key and needs to be removed.
+
+        This modifies the crash in-place.
+
+        :arg dict raw_crash: raw crash data
+
+        """
+        if not raw_crash:
+            return
+
+        for key in list(raw_crash.keys()):
```

Comment:
Ha! This is (negative) effect of code reviewing. I read it line by line and stopped as soon as I saw that line. Had I read 3 more lines down I would have understood the why. 
Just goes to show how it's easy to get caught up in something (that stands out) quickly when reading code. 
Yeah, something like `for key in list(raw_crash.keys()):   # iterate over a *copy* of the keys` would work for me.